### ISSUE TITLE
[CT-799] [Bug] Non existing model should give an error not a warning

### ISSUE THREAD

**Issue by imanebosch** [dbt-labs/dbt-core#5433]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

If we execute the command `dbt run --select <model path>` and the model does not exist dbt raises a warning message:

```
14:25:40  The selection criterion '<model path>' does not match any nodes
14:25:40  [WARNING]: Nothing to do. Try checking your model configs and model specification args
```

In beta, when orchestrating the tasks in airflow the scheduler could continue the dag because the table was created manually from the local environment. When the dag was deployed in production, the next task failed and it was not clear why. 

### Expected Behavior

Raise an error rather than a WARNING

### Steps To Reproduce

Run dbt run --select <path of a model that does not exist>

### Relevant log output

```shell
14:25:40  The selection criterion '<model path>' does not match any nodes
14:25:40  [WARNING]: Nothing to do. Try checking your model configs and model specification args
```
```


### Environment

```markdown
- OS: slim-buster docker image
- Python: 3.8.10
- dbt: 1.0.8
```

### What database are you using dbt with?

redshift

**Comment by jtcohen6** [comment]
@imanebosch This is a case where some users may want dbt to return an error (for good reasons), and others may want dbt to return a warning (also for good reasons). For instance, imagine a CI build against a PR with no changes. If the CI job is running `dbt build -s state:modified`, and `state:modified` returns zero models, it would be pretty odd for that CI check to report back with a failure!

There are two approaches we can use to make this behavior configurable:
1. The `WARN_ERROR` config ([docs](https://docs.getdbt.com/reference/global-configs#warnings-as-errors)), which can be set via CLI flag (`--warn-error`), env var (`DBT_WARN_ERROR`), or in `profiles.yml` for your runtime environment (Airflow)
2. Structured logging, by providing a reliable programmatic way to identify and react to the presence of this specific warning message (`[WARNING]: Nothing to do. Try checking your model configs and model specification args`). Today, all warnings that respect the `WARN_ERROR` config are fired as the same event type, [`GeneralWarningMsg`](https://github.com/dbt-labs/dbt-core/blob/60e491b3c178ee8112dbe1311bcc69e5c778ccfa/core/dbt/events/types.py#L2419), with the same code (`Z046`). We have work tracked to delineate between those events: https://github.com/dbt-labs/dbt-core/issues/4836#issuecomment-1061003390

In summary, I think you have a reasonable mechanism to raise an error today, via `WARN_ERROR`, and we'll be prioritizing work around structured logging to better handle and disambiguate these conditional warnings/errors. In the meantime, I'm going to close this issue.
